### ISSUE TITLE
Valgrind: Conditional jump or move depends on uninitialised value in ob_digit[0] of PyLongObject

### ISSUE THREAD

**Issue by OveSo** [python/cpython#119022]
# Bug report

### Bug description:

There are several cases, where ob_digit[0] of PyLongObject will not be initialized before evaluation.
I fixed that issue in Python 3.11.8 with this patch - it also concerns Python 3.11.9:
```python
--- longobject.c_orig	2024-02-06 22:21:21.000000000 +0100
+++ longobject.c	2024-05-13 17:47:25.846011498 +0200
@@ -165,6 +165,7 @@
         return NULL;
     }
     _PyObject_InitVar((PyVarObject*)result, &PyLong_Type, size);
+    result->ob_digit[0] = 0;
     return result;
 }
```
For example, one case is in longobject.c:922 in _PyLong_FromByteArray() if numsignificantbytes==0.
First, I fixed it locally in _PyLong_FromByteArray() but than I found similar cases in other functions.
Therefore, I decided to initialize ob_digit[0] right in _PyLong_New() in order to catch all situations.
Kind regards,
Ove

### CPython versions tested on:

3.11

### Operating systems tested on:

Linux

**Comment by sweeneyde** [comment]
Just an initial comment: I'm assuming this was fixed in 3.12+ by https://github.com/python/cpython/issues/102509.

3.11 looks like it's only accepting security fixes now, but I'm not sure whether this qualifies or not.

**Comment by sobolevn** [comment]
Yes, this was fixed in https://github.com/python/cpython/pull/102510

3.11 is security only, so it was not backported.
3.12+ are fixed.
